src/hooks/index.test.jsx

```diff
@@ -205,13 +207,11 @@ describe.each([
   test('updates before layout effects', async () => {
     let current
     const TestMe = () => {
-      const [rerender, setRerender] = useState(false)
       const oneRef = useRef()
       useLayoutEffect(() => {
-        current = oneRef.current
-        setRerender(true)
-      })
-      const mergedRef = useX(rerender ? [oneRef] : [])
+        current = oneRef.current;
+      }, [])
+      const mergedRef = useX([oneRef])
```

Comment:
I am not totally sure "what" this the goal of this test, but proposed code is better matching it's name.

Reply:
The purpose of this test is specifically for _updates_ rather than _initialization_.

It's easy to for initialization to happen before layout effects, but depending on the approach to handling updates, especially if you put them inside a layout effect, it can be hard or impossible to ensure refs are fully updated before layout effects run.

In other words, this test shows when a hook relies on effects or layout effects for updating. Any hook that does that will fail this test. The reason that a hook might rely on effects for updating is to avoid making updates during render, which isn't allowed in concurrent mode.

So the test was correct as written. We need to render initially one way, then rerender with an update to the merged ref. The test checks whether the ref was properly updated before the inline layout effect.